fix(build): `uname` handling for mac M1 arch. Fixes #13112

Adrien-D opened the pull request:
<!-- markdownlint-disable MD041 -- this is rendered within existing HTML, so allow starting without an H1 -->

<!--

### Before you open your PR

- Run `make pre-commit -B` to fix codegen and lint problems (build will fail).
- [Signed-off your commits](https://github.com/apps/dco/) (otherwise the DCO check will fail).
- Used [a conventional commit message](https://www.conventionalcommits.org/en/v1.0.0/).

### When you open your PR

- PR title format should also conform to [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).
- "Fixes #" is in both the PR title (for release notes) and this description (to automatically link and close the issue).
- Create the PR as draft.
- Once builds are green, mark your PR "Ready for review".

When changes are requested, please address them and then dismiss the review to get it reviewed again.

-->

<!-- Does this PR fix an issue -->

Fixes #13112

### Motivation

<!-- TODO: Say why you made your changes. -->
Unable to build the project using `make start UI=true`, and fails because of my M1 arch

### Modifications

<!-- TODO: Say what changes you made. -->
I updated the `TARGET_PLATFORM` value on the Makefile to be able to also handle my arch value which is `aarch64`
<!-- TODO: Attach screenshots if you changed the UI. -->

### Verification

<!-- TODO: Say how you tested your changes. -->
![image](https://github.com/argoproj/argo-workflows/assets/4726647/e492ae44-3028-4351-87dc-df51c99a8754)
<!--
### Beyond this PR

Thank you for submitting this! Have you ever thought of becoming a Reviewer or Approver on the project?

Argo Workflows is seeking more community involvement and ultimately more [Reviewers and Approvers](https://github.com/argoproj/argoproj/blob/main/community/membership.md) to help keep it viable.
See [Sustainability Effort](https://github.com/argoproj/argo-workflows/blob/main/community/sustainability_effort.md) for more information.

-->

agilgur5 commented on Makefile:
```diff
@@ -25,7 +25,13 @@ SRC                   := $(GOPATH)/src/github.com/argoproj/argo-workflows
 IMAGE_NAMESPACE       ?= quay.io/argoproj
 DEV_IMAGE             ?= $(shell [ `uname -s` = Darwin ] && echo true || echo false)
-TARGET_PLATFORM := $(shell [ `uname -m` = arm64 ] && echo linux/arm64 || echo linux/amd64)
+PLATFORM ?= $(shell uname -m)
+ifeq ($(PLATFORM), x86_64)
+PLATFORM=amd64
+else ifeq ($(PLATFORM), arm)
+PLATFORM=arm64
+endif
+TARGET_PLATFORM := $(shell echo linux/${PLATFORM})
```
```suggestion
TARGET_PLATFORM       ?= $(shell [ `uname -m` = x86_64 ] && echo linux/amd64 || echo linux/amd64)
```

can't we just flip the conditional? also `aarch64` and `arm64` are [equivalent](https://stackoverflow.com/a/47274698/3431180).

(plus change to `?=` and align spacing)

otherwise it might be more resilient to use `go env GOARCH`:
```suggestion
TARGET_PLATFORM       ?= linux/$(shell go env GOARCH)
```

and that would match https://github.com/argoproj/argo-workflows/pull/12754#pullrequestreview-1982586495 et al

Adrien-D commented:
@agilgur5 I implemented your last option as it looks way nicer

Also I was facing an issue running `make pre-commit -B` [described here](https://github.com/nektos/act/issues/1853#issue-1750881543) , so I implemented [the recommended fix](https://github.com/nektos/act/issues/1853#issuecomment-1620853901)

```
vscode ➜ ~/go/src/github.com/argoproj/argo-workflows (fix-make-mac-m1) $ mkdocs build --strict
Traceback (most recent call last):
  File "/home/vscode/.local/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/vscode/.local/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/vscode/.local/lib/python3.10/site-packages/mkdocs/__main__.py", line 281, in build_command
    cfg = config.load_config(**kwargs)
  File "/home/vscode/.local/lib/python3.10/site-packages/mkdocs/config/base.py", line 362, in load_config
    from mkdocs.config.defaults import MkDocsConfig
  File "/home/vscode/.local/lib/python3.10/site-packages/mkdocs/config/defaults.py", line 7, in <module>
    from mkdocs.config import config_options as c
  File "/home/vscode/.local/lib/python3.10/site-packages/mkdocs/config/config_options.py", line 36, in <module>
    from mkdocs import plugins, theme, utils
  File "/home/vscode/.local/lib/python3.10/site-packages/mkdocs/theme.py", line 16, in <module>
    from mkdocs import localization, utils
  File "/home/vscode/.local/lib/python3.10/site-packages/mkdocs/localization.py", line 16, in <module>
    from babel.support import NullTranslations, Translations
  File "/home/vscode/.local/lib/python3.10/site-packages/babel/support.py", line 23, in <module>
    from babel.dates import format_date, format_datetime, format_time, format_timedelta
  File "/home/vscode/.local/lib/python3.10/site-packages/babel/dates.py", line 34, in <module>
    from babel import localtime
  File "/home/vscode/.local/lib/python3.10/site-packages/babel/localtime/__init__.py", line 41, in <module>
    LOCALTZ = get_localzone()
  File "/home/vscode/.local/lib/python3.10/site-packages/babel/localtime/__init__.py", line 37, in get_localzone
    return _get_localzone()
  File "/home/vscode/.local/lib/python3.10/site-packages/babel/localtime/_unix.py", line 49, in _get_localzone
    tzinfo = _get_tzinfo(zone_name)
  File "/home/vscode/.local/lib/python3.10/site-packages/babel/localtime/_helpers.py", line 21, in _get_tzinfo
    return zoneinfo.ZoneInfo(tzenv)
  File "/usr/lib/python3.10/zoneinfo/_tzpath.py", line 67, in find_tzfile
    _validate_tzfile_path(key)
  File "/usr/lib/python3.10/zoneinfo/_tzpath.py", line 81, in _validate_tzfile_path
    raise ValueError(
ValueError: ZoneInfo keys may not be absolute paths, got: /UTC
```

agilgur5 commented:
Yea I got the `ZoneInfo` error recently too and upvoted that workaround earlier 😅  -- there's almost certainly some dep somewhere causing that (which would not be the first time I've had a random TZ build error pop up in Python due to deps). I do have a branch to add a Poetry config and `poetry.lock` to the repo that may help prevent such a thing.

agilgur5 reviewed (approved):
LGTM now, thanks for the fixes

agilgur5 commented:
Ah hitting the DCO issue https://github.com/dcoapp/app/issues/211; it was _just_ resolved, needs a new commit to re-trigger
